app/src/components/testCaseList/testCaseList.tsx

```diff
@@ -0,0 +1,152 @@
+  searchValue?: string;
+  onSearchChange?: (value: string) => void;
+  onEdit: () => void;
+  onDelete: (testCaseId: string | number) => void;
+  onDuplicate: (testCase: TestCase) => void;
+  onMove: () => void;
+}
+
+export const TestCaseList = memo(
+  ({
+    testCases = mockTestCases,
+    loading = false,
+    currentPage = DEFAULT_CURRENT_PAGE,
+    itemsPerPage,
+    searchValue = '',
+    onSearchChange,
+    onEdit,
+    onDelete,
+    onDuplicate,
+    onMove,
+  }: TestCaseListProps) => {
+    const { formatMessage } = useIntl();
+
+    const startIndex = (currentPage - 1) * itemsPerPage;
+    const endIndex = startIndex + itemsPerPage;
+    const currentData = testCases.slice(startIndex, endIndex);
+
+    if (loading) {
+      return (
+        <div className={cx('test-case-list', 'loading')}>
+          <div className={cx('loading-message')}>{formatMessage(messages.loadingMessage)}</div>
+        </div>
+      );
+    }
+
+    const tableData = currentData.map((testCase) => ({
+      id: testCase.id,
+      name: {
+        content: testCase.name,
+        component: (
+          <TestCaseNameCell status={testCase.status} name={testCase.name} tags={testCase.tags} />
+        ),
+      },
+      lastExecution: {
+        content: testCase.lastExecution,
+        component: (
+          <TestCaseExecutionCell
+            lastExecution={testCase.lastExecution}
+            onEdit={() => onEdit()}
+            onDelete={() => onDelete(testCase.id)}
+            onDuplicate={() => onDuplicate(testCase)}
+            onMove={() => onMove()}
+          />
+        ),
+      },
+    }));
+
+    const primaryColumn = {
+      key: 'name',
+      header: formatMessage(messages.nameHeader),
+      width: 'auto',
+      align: 'left' as const,
+    };
+
+    const fixedColumns = [
+      {
+        key: 'lastExecution',
+        header: formatMessage(messages.executionHeader),
+        width: 164,
+        align: 'left' as const,
+      },
+    ];
+
+    return (
+      <div className={cx('test-case-list')}>
+        <div className={cx('controls')}>
+          <div className={cx('controls-title')}>{formatMessage(messages.allTestCasesTitle)}</div>
+          <div className={cx('controls-actions')}>
+            <div className={cx('search-section')}>
+              <SearchField
+                isLoading={loading}
+                searchValue={searchValue}
+                setSearchValue={onSearchChange}
+                onFilterChange={onSearchChange}
+                placeholder={formatMessage(messages.searchPlaceholder)}
+              />
+              <div className={cx('filter-icon')}>
+                <FilterOutlineIcon />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        {currentData.length > 0 ? (
+          <Table
+            data={tableData}
+            fixedColumns={fixedColumns}
+            primaryColumn={primaryColumn}
+            sortingColumn={undefined}
+            sortableColumns={[]}
```

Comment:
Let's make them optional and not provide at all in this case

Reply:
removed